go-controller/pkg/config/config.go

```diff
@@ -1542,19 +1543,17 @@ func initConfigWithPath(ctx *cli.Context, exec kexec.Interface, saPath string, d
 		retConfigFile = configFile
 	}
 
-	f, err := os.Open(configFile)
+	iniCfg, err := ini.Load(configFile)
 	// Failure to find a default config file is not a hard error
 	if err != nil && !configFileIsDefault {
 		return "", fmt.Errorf("failed to open config file %s: %v", configFile, err)
 	}
-	if f != nil {
-		defer f.Close()
-
+	if iniCfg != nil {
 		// Parse ovn-k8s config file.
-		if err = gcfg.ReadInto(&cfg, f); err != nil {
-			return "", fmt.Errorf("failed to parse config file %s: %v", f.Name(), err)
+		if err = iniCfg.StrictMapTo(&cfg); err != nil {
```

Comment:
The problem with ignoring unknown fields with not even a warning it that the user is never gonna know why the change isn't taking effect. We need to at least log a warning that says what got ignored? 

Also from reading https://pkg.go.dev/gopkg.in/ini.v1#StrictMapTo , "which returns all possible error including value parsing error." - all possible errors? - wonder what all errors this catches. Does it ignore "unknown sections" as well or just "unknown keys"?

Reply:
This is my concern as well here (not having warnings to the user in case of unknown fields). But the `StrictMapTo` method uses the struct (and not the config file as source) for the mapping. That way it is not aware of additional fields in the file. I can only return an error in case of a type mismatch (when using the strict mode).